File: src/main/java/com/example/cqrs/query/service/AccountQueryUseCase.java
package com.example.cqrs.query.service;

import com.example.cqrs.query.entity.AccountView;
import com.example.cqrs.command.entity.event.AbstractAccountEvent;

import java.util.List;

public interface AccountQueryUseCase {

    // 계좌 조회
    AccountView getAccount(String accountId);

    // 계좌 이력 조회
    List<AbstractAccountEvent> getAccountHistory(String accountId);

    // 사용자 거래 이력 조회
    List<AbstractAccountEvent> getUserTransactions(String userId);

    // 연관 거래 조회
    List<AbstractAccountEvent> getRelatedTransactions(String correlationId);

    // 활성 계좌 조회 메서드 추가
    List<String> getActiveAccountIds();

}

File: src/main/java/com/example/cqrs/application/account/command/service/usecase/AccountQueryUseCase.kt
package com.example.cqrs.application.account.command.service.usecase

import com.example.cqrs.infrastructure.eventstore.entity.base.AccountEventBaseEntity
import com.example.cqrs.infrastructure.persistence.command.entity.AccountEntity

interface AccountQueryUseCase {
    fun getAccount(accountId: String): AccountEntity
    fun getAccountHistory(accountId: String): List<AccountEventBaseEntity>
    fun getUserTransactions(userId: String): List<AccountEventBaseEntity>
    fun getRelatedTransactions(correlationId: String): List<AccountEventBaseEntity>
    fun getActiveAccountIds(): List<String>
}